app/controllers/foods_controller.rb

```diff
@@ -0,0 +1,70 @@
+class FoodsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_food, only: %i[show edit update destroy]
+
+  # GET /foods or /foods.json
+  def index
+    @foods = Food.all
```

Comment:
As in requirements it is written there that the current user should only see foods created by him so please add here

`@foods = Food.where(user: current_user)`